packages/tendermint-rpc/src/client.ts

```diff
@@ -248,14 +249,47 @@ export class Client {
     };
   }
 
-  public async validators(height?: number): Promise<responses.ValidatorsResponse> {
+  public async validators(params: requests.ValidatorsParams): Promise<responses.ValidatorsResponse> {
     const query: requests.ValidatorsRequest = {
       method: requests.Method.Validators,
-      params: { height: height },
+      params: params,
     };
     return this.doCall(query, this.p.encodeValidators, this.r.decodeValidators);
   }
 
+  public async validatorsAll(params: requests.ValidatorsParams): Promise<responses.ValidatorsResponse> {
+    let page = params.page || 1;
+    const validators: responses.Validator[] = [];
+    const baseParams = {
+      per_page: 50,
+      height: params.height ?? (await this.status()).syncInfo.latestBlockHeight,
+      ...params,
+    };
+    let done = false;
+    let blockHeight = 0;
```

Comment:
Hmmm... if this was undefined or number, we could use this instead of querying.
Whatever the first one requested, keep using it.

```
let page = params.page || 1;
let blockHeight = params.height;
let perPage = params.perPage || 50;

while (!done) {
    const resp = await this.validators({
      page: page,
      height: blockHeight,
      perPage: perPage,
    });
    // ??? only set if blockHeight was undefined before?
    blockHeight = resp.blockHeight;
}
```

Reply:
This also removes the need for the status call (the first response will give us the most recent validator set if height was undefined)